2000-08-03  Ulrich Drepper  <drepper@redhat.com>

	* pthread.c (__pthread_set_own_extricate_if): Remove locking.

--- glibc-2.1.3/linuxthreads/pthread.c.jj	Wed Mar 22 22:26:01 2000
+++ glibc-2.1.3/linuxthreads/pthread.c	Fri Aug 25 09:37:26 2000
@@ -775,9 +775,7 @@ weak_alias (__pthread_getconcurrency, pt
 
 void __pthread_set_own_extricate_if(pthread_descr self, pthread_extricate_if *peif)
 {
-  __pthread_lock(self->p_lock, self);
   THREAD_SETMEM(self, p_extricate, peif);
-  __pthread_unlock(self->p_lock);
 }
 
 /* Primitives for controlling thread execution */
